src/cpu/core_dyn_x86/decoder.h

```diff
@@ -2900,6 +2905,9 @@ static CacheBlock * CreateCacheBlock(CodePageHandler * codepage,PhysPt start,Bit
 finish_block:
 	/* Setup the correct end-address */
 	decode.active_block->page.end=--decode.page.index;
-//	LOG_MSG("Created block size %d start %d end %d",decode.block->cache.size,decode.block->page.start,decode.block->page.end);
+	dyn_mem_execute(cache_addr, cache_bytes);
+	dyn_cache_invalidate(cache_addr, cache_bytes);
```

Comment:
Looking at all the `dyn_mem_execute(...)` calls, it looks like we always invalidate. 

**git grep -A1 'dyn_mem_execute(cache_addr, cache_bytes)'**
```
src/cpu/core_dyn_x86/decoder.h: dyn_mem_execute(cache_addr, cache_bytes);
src/cpu/core_dyn_x86/decoder.h- dyn_cache_invalidate(cache_addr, cache_bytes);
--
src/cpu/core_dyn_x86/risc_x64.h:        dyn_mem_execute(cache_addr, cache_bytes);
src/cpu/core_dyn_x86/risc_x64.h-        dyn_cache_invalidate(cache_addr, cache_bytes);
--
src/cpu/core_dyn_x86/risc_x64.h:        dyn_mem_execute(cache_addr, cache_bytes);
src/cpu/core_dyn_x86/risc_x64.h-        dyn_cache_invalidate(cache_addr, cache_bytes);
--
src/cpu/core_dynrec/decoder.h:  dyn_mem_execute(cache_addr, cache_bytes);
src/cpu/core_dynrec/decoder.h-  dyn_cache_invalidate(cache_addr, cache_bytes);
--
src/cpu/dyn_cache.h:            dyn_mem_execute(cache_addr, cache_bytes);
src/cpu/dyn_cache.h-            dyn_cache_invalidate(cache_addr, cache_bytes);
```

Maybe `dyn_mem_execute` can call `dyn_cache_invalidate` internally, and strongly tie this sequence together? (asking because maybe it doesn't make sense in some future cases?)